### PR TITLE
feat(cilium): adding EnableLocalRedirectPolicy parameter

### PR DESCRIPTION
**Pull request by ana-aguilar7**
## Context

This PR adds the EnableLocalRedirectPolicy parameter to the Cilium spec in kOps, allowing optional activation of the Local Redirect Policy.

### CODE REVIEW

File: pkg/model/components/cilium.go [hunk 1]
```diff
@@ -139,6 +139,10 @@ func (b *CiliumOptionsBuilder) BuildOptions(o *kops.Cluster) error {
 		c.EnableL7Proxy = fi.PtrTo(true)
 	}
 
+	if c.EnableLocalRedirectPolicy == nil {
+		c.EnableLocalRedirectPolicy = fi.PtrTo(true)
```
Comment (hakman) [inline pkg/model/components/cilium.go:143]: This makes the default `true`. Not sure if we want this as default.

Comment (hakman) [reply pkg/model/components/cilium.go]: You can just remove any changes to the file and run `hack/update-expected.sh`.

**Comment by ana-aguilar7** [comment]
@hakmancould you help me? There's an error happening in the tests, and I'm not sure what's causing it.
It looks related to the test environment itself

**Comment by hakman** [comment]
@ana-aguilar7 Please also run `make apimachinery` and commit the result.

File: pkg/model/components/cilium.go [hunk 2]
```diff
@@ -139,6 +139,10 @@ func (b *CiliumOptionsBuilder) BuildOptions(o *kops.Cluster) error {
 		c.EnableL7Proxy = fi.PtrTo(true)
 	}
 
+	if c.EnableLocalRedirectPolicy == nil {
+		c.EnableLocalRedirectPolicy = fi.PtrTo(false)
+	}
+
```
Comment (hakman) [inline pkg/model/components/cilium.go:145]: Please remove this change. It should not affect any of the output.

Comment (ana-aguilar7) [reply pkg/model/components/cilium.go]: @hakman I made a mistake I will create another PR and fix it

Comment (hakman) [reply pkg/model/components/cilium.go]: no worries. looking forward to it.

**Pull request closed without merging**